Telegram Bot

sw1pr0g opened the pull request.

sw1pr0g commented:
I’ve just pushed the changes. The basic bot implementation is done, it starts up and logs are being written. The /start button throws an error and needs to be implemented (locales need to be configured first, and user data should be collected upon the first bot launch)

F1st3K reviewed (changes requested):
containerization needs to be added (docker)

otherwise everything is fine

sw1pr0g commented:
@F1st3K i've updated branch, added containerization, but it's not build on my mac M1

stack trace:
```
Dockerfile:14
--------------------
  13 |     
  14 | >>> RUN pip install --no-cache-dir "poetry==$POETRY_VERSION" \
  15 | >>>     && poetry install --without admin --without dev --no-root \
  16 | >>>     && pip uninstall -y poetry \
  17 | >>>     && pybabel compile -d bot/locales \
  18 | >>>     && rm -rf /home/appuser/.cache \
  19 | >>>     && rm -rf $POETRY_CACHE_DIR \
  20 | >>>     && adduser -D appuser \
  21 | >>>     && chown -R appuser:appuser .
  22 |     
--------------------
ERROR: failed to solve: process "/bin/sh -c pip install --no-cache-dir \"poetry==$POETRY_VERSION\"     && poetry install --without admin --without dev --no-root     && pip uninstall -y poetry     && pybabel compile -d bot/locales     && rm -rf /home/appuser/.cache     && rm -rf $POETRY_CACHE_DIR     && adduser -D appuser     && chown -R appuser:appuser ." did not complete successfully: exit code: 1

View build details: docker-desktop://dashboard/build/desktop-linux/desktop-linux/ve2iide0gjb1wqaz4vuegtza4
```

can you test it on your PC?

F1st3K reviewed (changes requested):
Identical error:

```bash
┌─(~/Repos/EasyVPN/services/TelegramBot)────────────────────────────────────────────────────────────────────────────(f1st3k@localhost:pts/0)─┐
└─(12:02:37 on feature/telegram-bot)──> docker build -t easyvpn/bot ./                                                   125 ↵ ──(Thu,Apr03)─┘
[+] Building 90.2s (8/8) FINISHED                                                                                               docker:default
 => [internal] load build definition from Dockerfile                                                                                      0.1s
 => => transferring dockerfile: 676B                                                                                                      0.0s
 => [internal] load metadata for docker.io/library/python:3.12.8-alpine3.21                                                               2.7s
 => [internal] load .dockerignore                                                                                                         0.1s
 => => transferring context: 127B                                                                                                         0.0s
 => [1/4] FROM docker.io/library/python:3.12.8-alpine3.21@sha256:ba13ef990f6e5d13014e9e8d04c02a8fdb0fe53d6dccf6e19147f316e6cc3a84         3.6s
 => => resolve docker.io/library/python:3.12.8-alpine3.21@sha256:ba13ef990f6e5d13014e9e8d04c02a8fdb0fe53d6dccf6e19147f316e6cc3a84         0.1s
 => => sha256:7e3e920185fe2394c1f10fe5a76321ab8788fe2c3bcc0103ec80e0a6befd8804 1.73kB / 1.73kB                                            0.0s
 => => sha256:ba13ef990f6e5d13014e9e8d04c02a8fdb0fe53d6dccf6e19147f316e6cc3a84 9.02kB / 9.02kB                                            0.0s
 => => sha256:1d654ef69d39f104f3982d7ef1f5644e52dcbb04991b822f2701e40a69f164a9 5.10kB / 5.10kB                                            0.0s
 => => sha256:2bd2a30403bed8ce2d8f998b1be5032cb97e2f9f46ec26c039d29bf59b3819c1 458.75kB / 458.75kB                                        0.7s
 => => sha256:b2d05111962e5378328cf22c62e3a83da84cf90592539e8248925ec409f87af5 13.61MB / 13.61MB                                          2.4s
 => => sha256:83e8891da8f55682d107f9ba00ba9688a1342a468ee8f1953ef38caae2f042d7 249B / 249B                                                0.3s
 => => extracting sha256:2bd2a30403bed8ce2d8f998b1be5032cb97e2f9f46ec26c039d29bf59b3819c1                                                 0.1s
 => => extracting sha256:b2d05111962e5378328cf22c62e3a83da84cf90592539e8248925ec409f87af5                                                 0.4s
 => => extracting sha256:83e8891da8f55682d107f9ba00ba9688a1342a468ee8f1953ef38caae2f042d7                                                 0.0s
 => [internal] load build context                                                                                                         0.3s
 => => transferring context: 135.41kB                                                                                                     0.0s
 => [2/4] WORKDIR /usr/src/app                                                                                                            0.7s
 => [3/4] COPY . .                                                                                                                        0.2s
 => ERROR [4/4] RUN pip install --no-cache-dir "poetry==2.0.1"     && poetry install --without admin --without dev --no-root     && pip  82.6s
------                                                                                                                                         
 > [4/4] RUN pip install --no-cache-dir "poetry==2.0.1"     && poetry install --without admin --without dev --no-root     && pip uninstall -y poetry     && pybabel compile -d bot/locales     && rm -rf /home/appuser/.cache     && rm -rf /tmp/poetry_cache     && adduser -D appuser     && chown -R appuser:appuser .:                                                                                                                   
3.717 Collecting poetry==2.0.1                                                                                                                 
18.91   WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ReadTimeoutError("HTTPSConnectionPool(host='files.pythonhosted.org', port=443): Read timed out. (read timeout=15)")': /packages/da/7b/8fb18c676e360e7b9ded8fa09c777b55ff2995bc4d34a542b5d13edf499d/poetry-2.0.1-py3-none-any.whl.metadata
34.56   WARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ReadTimeoutError("HTTPSConnectionPool(host='files.pythonhosted.org', port=443): Read timed out. (read timeout=15)")': /packages/da/7b/8fb18c676e360e7b9ded8fa09c777b55ff2995bc4d34a542b5d13edf499d/poetry-2.0.1-py3-none-any.whl.metadata
50.71   WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ReadTimeoutError("HTTPSConnectionPool(host='files.pythonhosted.org', port=443): Read timed out. (read timeout=15)")': /packages/da/7b/8fb18c676e360e7b9ded8fa09c777b55ff2995bc4d34a542b5d13edf499d/poetry-2.0.1-py3-none-any.whl.metadata
51.11   Downloading poetry-2.0.1-py3-none-any.whl.metadata (7.0 kB)
51.27 Collecting build<2.0.0,>=1.2.1 (from poetry==2.0.1)
51.36   Downloading build-1.2.2.post1-py3-none-any.whl.metadata (6.5 kB)
51.54 Collecting cachecontrol<0.15.0,>=0.14.0 (from cachecontrol[filecache]<0.15.0,>=0.14.0->poetry==2.0.1)
51.72   Downloading cachecontrol-0.14.2-py3-none-any.whl.metadata (3.1 kB)
51.97 Collecting cleo<3.0.0,>=2.1.0 (from poetry==2.0.1)
52.11   Downloading cleo-2.1.0-py3-none-any.whl.metadata (12 kB)
52.53 Collecting dulwich<0.23.0,>=0.22.6 (from poetry==2.0.1)
52.62   Downloading dulwich-0.22.8-py3-none-any.whl.metadata (4.9 kB)
52.74 Collecting fastjsonschema<3.0.0,>=2.18.0 (from poetry==2.0.1)
52.83   Downloading fastjsonschema-2.21.1-py3-none-any.whl.metadata (2.2 kB)
52.94 Collecting installer<0.8.0,>=0.7.0 (from poetry==2.0.1)
53.03   Downloading installer-0.7.0-py3-none-any.whl.metadata (936 bytes)
53.18 Collecting keyring<26.0.0,>=25.1.0 (from poetry==2.0.1)
53.27   Downloading keyring-25.6.0-py3-none-any.whl.metadata (20 kB)
53.46 Collecting packaging>=24.0 (from poetry==2.0.1)
53.55   Downloading packaging-24.2-py3-none-any.whl.metadata (3.2 kB)
53.66 Collecting pkginfo<2.0,>=1.12 (from poetry==2.0.1)
53.75   Downloading pkginfo-1.12.1.2-py3-none-any.whl.metadata (13 kB)
53.90 Collecting platformdirs<5,>=3.0.0 (from poetry==2.0.1)
53.99   Downloading platformdirs-4.3.7-py3-none-any.whl.metadata (11 kB)
54.17 Collecting poetry-core==2.0.1 (from poetry==2.0.1)
54.26   Downloading poetry_core-2.0.1-py3-none-any.whl.metadata (3.5 kB)
54.37 Collecting pyproject-hooks<2.0.0,>=1.0.0 (from poetry==2.0.1)
54.46   Downloading pyproject_hooks-1.2.0-py3-none-any.whl.metadata (1.3 kB)
54.59 Collecting requests<3.0,>=2.26 (from poetry==2.0.1)
54.69   Downloading requests-2.32.3-py3-none-any.whl.metadata (4.6 kB)
54.80 Collecting requests-toolbelt<2.0.0,>=1.0.0 (from poetry==2.0.1)
54.89   Downloading requests_toolbelt-1.0.0-py2.py3-none-any.whl.metadata (14 kB)
55.03 Collecting shellingham<2.0,>=1.5 (from poetry==2.0.1)
55.13   Downloading shellingham-1.5.4-py2.py3-none-any.whl.metadata (3.5 kB)
55.29 Collecting tomlkit<1.0.0,>=0.11.4 (from poetry==2.0.1)
55.38   Downloading tomlkit-0.13.2-py3-none-any.whl.metadata (2.7 kB)
55.51 Collecting trove-classifiers>=2022.5.19 (from poetry==2.0.1)
55.60   Downloading trove_classifiers-2025.3.19.19-py3-none-any.whl.metadata (2.3 kB)
55.82 Collecting virtualenv<21.0.0,>=20.26.6 (from poetry==2.0.1)
55.92   Downloading virtualenv-20.30.0-py3-none-any.whl.metadata (4.5 kB)
56.13 Collecting msgpack<2.0.0,>=0.5.2 (from cachecontrol<0.15.0,>=0.14.0->cachecontrol[filecache]<0.15.0,>=0.14.0->poetry==2.0.1)
56.22   Downloading msgpack-1.1.0-cp312-cp312-musllinux_1_2_x86_64.whl.metadata (8.4 kB)
56.39 Collecting filelock>=3.8.0 (from cachecontrol[filecache]<0.15.0,>=0.14.0->poetry==2.0.1)
56.48   Downloading filelock-3.18.0-py3-none-any.whl.metadata (2.9 kB)
56.59 Collecting crashtest<0.5.0,>=0.4.1 (from cleo<3.0.0,>=2.1.0->poetry==2.0.1)
56.68   Downloading crashtest-0.4.1-py3-none-any.whl.metadata (1.1 kB)
57.78 Collecting rapidfuzz<4.0.0,>=3.0.0 (from cleo<3.0.0,>=2.1.0->poetry==2.0.1)
57.87   Downloading rapidfuzz-3.12.2-cp312-cp312-musllinux_1_2_x86_64.whl.metadata (12 kB)
58.07 Collecting urllib3>=1.25 (from dulwich<0.23.0,>=0.22.6->poetry==2.0.1)
58.16   Downloading urllib3-2.3.0-py3-none-any.whl.metadata (6.5 kB)
58.29 Collecting SecretStorage>=3.2 (from keyring<26.0.0,>=25.1.0->poetry==2.0.1)
58.39   Downloading SecretStorage-3.3.3-py3-none-any.whl.metadata (4.0 kB)
58.50 Collecting jeepney>=0.4.2 (from keyring<26.0.0,>=25.1.0->poetry==2.0.1)
58.60   Downloading jeepney-0.9.0-py3-none-any.whl.metadata (1.2 kB)
58.71 Collecting jaraco.classes (from keyring<26.0.0,>=25.1.0->poetry==2.0.1)
58.81   Downloading jaraco.classes-3.4.0-py3-none-any.whl.metadata (2.6 kB)
58.95 Collecting jaraco.functools (from keyring<26.0.0,>=25.1.0->poetry==2.0.1)
59.04   Downloading jaraco.functools-4.1.0-py3-none-any.whl.metadata (2.9 kB)
59.17 Collecting jaraco.context (from keyring<26.0.0,>=25.1.0->poetry==2.0.1)
59.27   Downloading jaraco.context-6.0.1-py3-none-any.whl.metadata (4.1 kB)
59.95 Collecting charset-normalizer<4,>=2 (from requests<3.0,>=2.26->poetry==2.0.1)
60.04   Downloading charset_normalizer-3.4.1-cp312-cp312-musllinux_1_2_x86_64.whl.metadata (35 kB)
60.23 Collecting idna<4,>=2.5 (from requests<3.0,>=2.26->poetry==2.0.1)
60.32   Downloading idna-3.10-py3-none-any.whl.metadata (10 kB)
60.49 Collecting certifi>=2017.4.17 (from requests<3.0,>=2.26->poetry==2.0.1)
60.58   Downloading certifi-2025.1.31-py3-none-any.whl.metadata (2.5 kB)
60.74 Collecting distlib<1,>=0.3.7 (from virtualenv<21.0.0,>=20.26.6->poetry==2.0.1)
60.83   Downloading distlib-0.3.9-py2.py3-none-any.whl.metadata (5.2 kB)
61.23 Collecting cryptography>=2.0 (from SecretStorage>=3.2->keyring<26.0.0,>=25.1.0->poetry==2.0.1)
61.32   Downloading cryptography-44.0.2-cp39-abi3-musllinux_1_2_x86_64.whl.metadata (5.7 kB)
61.45 Collecting more-itertools (from jaraco.classes->keyring<26.0.0,>=25.1.0->poetry==2.0.1)
61.54   Downloading more_itertools-10.6.0-py3-none-any.whl.metadata (37 kB)
61.89 Collecting cffi>=1.12 (from cryptography>=2.0->SecretStorage>=3.2->keyring<26.0.0,>=25.1.0->poetry==2.0.1)
61.99   Downloading cffi-1.17.1-cp312-cp312-musllinux_1_1_x86_64.whl.metadata (1.5 kB)
62.10 Collecting pycparser (from cffi>=1.12->cryptography>=2.0->SecretStorage>=3.2->keyring<26.0.0,>=25.1.0->poetry==2.0.1)
62.20   Downloading pycparser-2.22-py3-none-any.whl.metadata (943 bytes)
62.31 Downloading poetry-2.0.1-py3-none-any.whl (254 kB)
62.89 Downloading poetry_core-2.0.1-py3-none-any.whl (544 kB)
63.66    ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 544.8/544.8 kB 811.5 kB/s eta 0:00:00
63.76 Downloading build-1.2.2.post1-py3-none-any.whl (22 kB)
63.87 Downloading cachecontrol-0.14.2-py3-none-any.whl (21 kB)
63.98 Downloading cleo-2.1.0-py3-none-any.whl (78 kB)
64.15 Downloading dulwich-0.22.8-py3-none-any.whl (273 kB)
64.52 Downloading fastjsonschema-2.21.1-py3-none-any.whl (23 kB)
64.63 Downloading installer-0.7.0-py3-none-any.whl (453 kB)
65.10 Downloading keyring-25.6.0-py3-none-any.whl (39 kB)
65.21 Downloading packaging-24.2-py3-none-any.whl (65 kB)
65.34 Downloading pkginfo-1.12.1.2-py3-none-any.whl (32 kB)
65.45 Downloading platformdirs-4.3.7-py3-none-any.whl (18 kB)
65.56 Downloading pyproject_hooks-1.2.0-py3-none-any.whl (10 kB)
65.65 Downloading requests-2.32.3-py3-none-any.whl (64 kB)
65.78 Downloading requests_toolbelt-1.0.0-py2.py3-none-any.whl (54 kB)
65.91 Downloading shellingham-1.5.4-py2.py3-none-any.whl (9.8 kB)
66.00 Downloading tomlkit-0.13.2-py3-none-any.whl (37 kB)
66.12 Downloading trove_classifiers-2025.3.19.19-py3-none-any.whl (13 kB)
66.22 Downloading virtualenv-20.30.0-py3-none-any.whl (4.3 MB)
68.15    ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.3/4.3 MB 2.3 MB/s eta 0:00:00
68.25 Downloading certifi-2025.1.31-py3-none-any.whl (166 kB)
68.38 Downloading charset_normalizer-3.4.1-cp312-cp312-musllinux_1_2_x86_64.whl (146 kB)
68.52 Downloading crashtest-0.4.1-py3-none-any.whl (7.6 kB)
68.61 Downloading distlib-0.3.9-py2.py3-none-any.whl (468 kB)
68.83 Downloading filelock-3.18.0-py3-none-any.whl (16 kB)
68.94 Downloading idna-3.10-py3-none-any.whl (70 kB)
69.05 Downloading jeepney-0.9.0-py3-none-any.whl (49 kB)
69.17 Downloading msgpack-1.1.0-cp312-cp312-musllinux_1_2_x86_64.whl (394 kB)
69.37 Downloading rapidfuzz-3.12.2-cp312-cp312-musllinux_1_2_x86_64.whl (4.6 MB)
70.25    ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.6/4.6 MB 5.2 MB/s eta 0:00:00
70.35 Downloading SecretStorage-3.3.3-py3-none-any.whl (15 kB)
70.45 Downloading urllib3-2.3.0-py3-none-any.whl (128 kB)
70.58 Downloading jaraco.classes-3.4.0-py3-none-any.whl (6.8 kB)
70.68 Downloading jaraco.context-6.0.1-py3-none-any.whl (6.8 kB)
70.78 Downloading jaraco.functools-4.1.0-py3-none-any.whl (10 kB)
70.87 Downloading cryptography-44.0.2-cp39-abi3-musllinux_1_2_x86_64.whl (4.3 MB)
71.70    ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.3/4.3 MB 5.1 MB/s eta 0:00:00
71.79 Downloading more_itertools-10.6.0-py3-none-any.whl (63 kB)
71.90 Downloading cffi-1.17.1-cp312-cp312-musllinux_1_1_x86_64.whl (488 kB)
72.07 Downloading pycparser-2.22-py3-none-any.whl (117 kB)
72.25 Installing collected packages: trove-classifiers, fastjsonschema, distlib, urllib3, tomlkit, shellingham, rapidfuzz, pyproject-hooks, pycparser, poetry-core, platformdirs, pkginfo, packaging, msgpack, more-itertools, jeepney, jaraco.context, installer, idna, filelock, crashtest, charset-normalizer, certifi, virtualenv, requests, jaraco.functools, jaraco.classes, dulwich, cleo, cffi, build, requests-toolbelt, cryptography, cachecontrol, SecretStorage, keyring, poetry
75.28 Successfully installed SecretStorage-3.3.3 build-1.2.2.post1 cachecontrol-0.14.2 certifi-2025.1.31 cffi-1.17.1 charset-normalizer-3.4.1 cleo-2.1.0 crashtest-0.4.1 cryptography-44.0.2 distlib-0.3.9 dulwich-0.22.8 fastjsonschema-2.21.1 filelock-3.18.0 idna-3.10 installer-0.7.0 jaraco.classes-3.4.0 jaraco.context-6.0.1 jaraco.functools-4.1.0 jeepney-0.9.0 keyring-25.6.0 more-itertools-10.6.0 msgpack-1.1.0 packaging-24.2 pkginfo-1.12.1.2 platformdirs-4.3.7 poetry-2.0.1 poetry-core-2.0.1 pycparser-2.22 pyproject-hooks-1.2.0 rapidfuzz-3.12.2 requests-2.32.3 requests-toolbelt-1.0.0 shellingham-1.5.4 tomlkit-0.13.2 trove-classifiers-2025.3.19.19 urllib3-2.3.0 virtualenv-20.30.0
75.29 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager, possibly rendering your system unusable.It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv. Use the --root-user-action option if you know what you are doing and want to suppress this warning.
81.68 Skipping virtualenv creation, as specified in config file.
81.87 
81.87 Group(s) not found: admin (via --without)
------
Dockerfile:14
--------------------
  13 |     
  14 | >>> RUN pip install --no-cache-dir "poetry==$POETRY_VERSION" \
  15 | >>>     && poetry install --without admin --without dev --no-root \
  16 | >>>     && pip uninstall -y poetry \
  17 | >>>     && pybabel compile -d bot/locales \
  18 | >>>     && rm -rf /home/appuser/.cache \
  19 | >>>     && rm -rf $POETRY_CACHE_DIR \
  20 | >>>     && adduser -D appuser \
  21 | >>>     && chown -R appuser:appuser .
  22 |     
--------------------
ERROR: failed to solve: process "/bin/sh -c pip install --no-cache-dir \"poetry==$POETRY_VERSION\"     && poetry install --without admin --without dev --no-root     && pip uninstall -y poetry     && pybabel compile -d bot/locales     && rm -rf /home/appuser/.cache     && rm -rf $POETRY_CACHE_DIR     && adduser -D appuser     && chown -R appuser:appuser ." did not complete successfully: exit code: 1
┌─(~/Repos/EasyVPN/services/TelegramBot)────────────────────────────────────────────────────────────────────────────(f1st3k@localhost:pts/0)─┐
└─(12:06:02 on feature/telegram-bot)──>                                                                                    1 ↵ ──(Thu,Apr03)─┘
```

sw1pr0g commented:
@F1st3K now all works fine

F1st3K reviewed (approved):
It's fine

F1st3K commented:
Documentation for deploy needs be added, suggest example:

```sh
docker build -t easyvpn/bot:local ./ 

docker run -d \ 
  --name bot \
  -e BOT_TOKEN=6618328943:AAEKMZYUJOliAEdBkibrhg8QMMy_xni-jEg \
  easyvpn/bot:local
```

F1st3K reviewed (changes requested):
So.. needs clarification:

write to bot: 
/start
- nothing

logs:

```sh
┌─(~/Repos/EasyVPN)─────────────────────────────────────────────────────────────────────────────────────────────────(f1st3k@localhost:pts/0)─┐
└─(19:24:08 on feature/telegram-bot)──> docker logs c183                                                                       ──(Thu,Apr03)─┘
2025-04-03 16:23:15.978 | INFO     | __main__:on_startup:13 - bot starting...
2025-04-03 16:23:16.738 | INFO     | __main__:on_startup:21 - Name     - -F1st3K
2025-04-03 16:23:16.739 | INFO     | __main__:on_startup:22 - Username - @F1st3K_bot
2025-04-03 16:23:16.739 | INFO     | __main__:on_startup:23 - ID       - 6618328943
2025-04-03 16:23:16.739 | INFO     | __main__:on_startup:31 - Groups Mode  - Enabled
2025-04-03 16:23:16.739 | INFO     | __main__:on_startup:32 - Privacy Mode - Enabled
2025-04-03 16:23:16.740 | INFO     | __main__:on_startup:33 - Inline Mode  - Disabled
2025-04-03 16:23:16.740 | INFO     | __main__:on_startup:35 - bot started
Cause exception while process update id=820181706 by bot id=6618328943
LookupError: I18n context is not set
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/site-packages/aiogram/dispatcher/dispatcher.py", line 309, in _process_update
    response = await self.feed_update(bot, update, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/aiogram/dispatcher/dispatcher.py", line 158, in feed_update
    response = await self.update.wrap_outer_middleware(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/aiogram/dispatcher/middlewares/error.py", line 25, in __call__
    return await handler(event, data)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/aiogram/dispatcher/middlewares/user_context.py", line 56, in __call__
    return await handler(event, data)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/aiogram/fsm/middleware.py", line 42, in __call__
    return await handler(event, data)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/aiogram/dispatcher/event/telegram.py", line 121, in trigger
    return await wrapped_inner(event, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/aiogram/dispatcher/event/handler.py", line 43, in call
    return await wrapped()
           ^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/aiogram/dispatcher/dispatcher.py", line 276, in _listen_update
    return await self.propagate_event(update_type=update_type, event=event, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/aiogram/dispatcher/router.py", line 146, in propagate_event
    return await observer.wrap_outer_middleware(_wrapped, event=event, data=kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/aiogram/dispatcher/router.py", line 141, in _wrapped
    return await self._propagate_event(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/aiogram/dispatcher/router.py", line 174, in _propagate_event
    response = await router.propagate_event(update_type=update_type, event=event, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/aiogram/dispatcher/router.py", line 146, in propagate_event
    return await observer.wrap_outer_middleware(_wrapped, event=event, data=kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/aiogram/dispatcher/router.py", line 141, in _wrapped
    return await self._propagate_event(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/aiogram/dispatcher/router.py", line 174, in _propagate_event
    response = await router.propagate_event(update_type=update_type, event=event, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/aiogram/dispatcher/router.py", line 146, in propagate_event
    return await observer.wrap_outer_middleware(_wrapped, event=event, data=kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/aiogram/dispatcher/router.py", line 141, in _wrapped
    return await self._propagate_event(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/aiogram/dispatcher/router.py", line 166, in _propagate_event
    response = await observer.trigger(event, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/aiogram/dispatcher/event/telegram.py", line 121, in trigger
    return await wrapped_inner(event, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/aiogram/dispatcher/event/handler.py", line 43, in call
    return await wrapped()
           ^^^^^^^^^^^^^^^
  File "/usr/src/bot/bot/handlers/start.py", line 13, in start_handler
    await message.answer(_("first message"), reply_markup=main_keyboard())
                         ^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/aiogram/utils/i18n/context.py", line 15, in gettext
    return get_i18n().gettext(*args, **kwargs)
           ^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/aiogram/utils/i18n/context.py", line 10, in get_i18n
    raise LookupError("I18n context is not set")
LookupError: I18n context is not set
┌─(~/Repos/EasyVPN)─────────────────────────────────────────────────────────────────────────────────────────────────(f1st3k@localhost:pts/0)─┐
└─(19:24:17 on feature/telegram-bot)──>  
```

sw1pr0g commented:
> I’ve just pushed the changes. The basic bot implementation is done, it starts up and logs are being written. The `/start` button throws an error and needs to be implemented (locales need to be configured first, and user data should be collected upon the first bot launch)

@F1st3K it's normal for now. i already wrote about this above

F1st3K reviewed (approved):
Ok, ty.
